rust/src/pet/coordinator.rs

```diff
@@ -211,27 +225,41 @@ impl Coordinator {
         encr_pk: &box_::PublicKey,
         dict_seed: &HashMap<box_::PublicKey, Vec<u8>>,
     ) -> Result<(), PetError> {
-        (dict_seed.len() == self.dict_sum.len()
-            && dict_seed.keys().all(|pk| self.dict_sum.contains_key(pk)))
-        .then_some({
-            dict_seed.iter().for_each(|(pk, seed)| {
-                self.dict_seed
-                    .get_mut(pk)
-                    .unwrap()
-                    .insert(*encr_pk, seed.clone());
-            });
-        })
-        .ok_or(PetError::InvalidMessage)
+        if dict_seed.len() != self.dict_sum.len() {
+            return Err(PetError::InvalidMessage);
+        }
+
+        if dict_seed.keys().all(|pk| self.dict_sum.contains_key(pk)) {
+            return Err(PetError::InvalidMessage);
+        }
```

Comment:
logic was reverted, should be `if !dict_seed.keys().all(|pk| self.dict_sum.contains_key(pk)) { return Err(...) }`

alternative, this can be rewritten to `if dict_seed.keys().any(|pk| !self.dict_sum.contains_key(pk)) { return Err(...) }`